system-addon/content-src/lib/snippets.js

```diff
@@ -3,6 +3,9 @@ const DATABASE_VERSION = 1;
 const SNIPPETS_OBJECTSTORE_NAME = "snippets";
 const SNIPPETS_UPDATE_INTERVAL_MS = 14400000; // 4 hours.
 
+const SnippetsEnabled = new Event("SnippetsEnabled");
```

Comment:
It seems a bit weird to me that you would create a single event ahead of time and re-dispatch the same one... maybe just store the type as a constants and create the event at dispatch time?